src/main/java/org/fogbowcloud/manager/core/threads/OpenProcessor.java

```diff
@@ -0,0 +1,159 @@
+package org.fogbowcloud.manager.core.threads;
+
+import java.util.Properties;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.constants.ConfigurationConstants;
+import org.fogbowcloud.manager.core.constants.DefaultConfigurationConstants;
+import org.fogbowcloud.manager.core.instanceprovider.InstanceProvider;
+import org.fogbowcloud.manager.core.models.linkedList.ChainedList;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+
+public class OpenProcessor implements Runnable {
+
+	private InstanceProvider localInstanceProvider;
+	private InstanceProvider remoteInstanceProvider;
+
+	private ChainedList openOrdersList;
+	private ChainedList pendingOrdersList;
+	private ChainedList failedOrdersList;
+	private ChainedList spawningOrdersList;
+
+	private String localMemberId;
+
+	/**
+	 * Attribute that represents the thread sleep time when there is no Orders
+	 * to be processed.
+	 */
+	private Long sleepTime;
+
+	private static final Logger LOGGER = Logger.getLogger(OpenProcessor.class);
+
+	public OpenProcessor(InstanceProvider localInstanceProvider, InstanceProvider remoteInstanceProvider,
+			String localMemberId, Properties properties) {
+		this.localInstanceProvider = localInstanceProvider;
+		this.remoteInstanceProvider = remoteInstanceProvider;
+		this.localMemberId = localMemberId;
+
+		// TODO: ChainedLists instanciation by Singleton pattern.
+
+		String schedulerPeriodStr = properties.getProperty(ConfigurationConstants.OPEN_ORDERS_SLEEP_TIME_KEY,
+				DefaultConfigurationConstants.OPEN_ORDERS_SLEEP_TIME);
+		this.sleepTime = Long.valueOf(schedulerPeriodStr);
+	}
+
+	@Override
+	public void run() {
+		while (true) {
+			try {
+				Order order = this.openOrdersList.getNext();
+				if (order != null) {
+					this.processOpenOrder(order);
+				} else {
+					LOGGER.info("There is no Open Order to be processed, sleeping OpenProcessor Thread...");
+					Thread.sleep(this.sleepTime);
+				}
+			} catch (Throwable e) {
+				LOGGER.error("Error while trying to Process an Open Order with the OpenProcessor Thread", e);
+			}
+		}
+	}
+
+	/**
+	 * Method that try to get an Instance for an Open Order. This method can
+	 * generate a race condition. For example: a user can delete a Open Order
+	 * while this method is trying to get an Instance for this Order.
+	 */
+	protected void processOpenOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+
+			if (orderState.equals(OrderState.OPEN)) {
+				LOGGER.info("Trying to get an Instance for Order [" + order.getId() + "]");
```

Comment:
replace Order by order in the log

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/2c530ad051c2f728874872ffbdc26a8b19dc9bd1).